File: atcoder.jp/contests/zone2021/tasks/zone2021_a/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const countText = (countString: string, s: string) =>
  [...s.matchAll(RegExp(countString, "g"))].length;

const main = () => {
  const S = input();

  const ans = countText("ZONe", S);
  console.log(ans);
};

main();

File: atcoder.jp/contests/zone2021/tasks/zone2021_a/main.cpp
#include <bits/stdc++.h>
using namespace std;

int count(const string &count_string, const string &s) {
  int ret = 0;
  for (size_t i = 0; i < s.size(); i++)
    if (s.substr(i, count_string.size()) == count_string) ret++;
  return ret;
}

int main() {
  string S;
  cin >> S;

  int ans = count("ZONe", S);
  cout << ans << endl;
}
